app_dart/dev/deploy.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+const String gcloudProjectIdFlag = 'project';
+const String gcloudProjectIdAbbrFlag = 'p';
+
+const String gcloudProjectVersionFlag = 'version';
+const String gcloudProjectVersionAbbrFlag = 'v';
+
+String gcloudProjectId;
+String gcloudProjectVersion;
+
+/// Check if [gcloudProjectIdFlag] and [gcloudProjectVersionFlag]
+/// were passed as arguments. If they were, also set [gcloudProjectId]
+/// and [gcloudProjectVersion] accordingly.
+bool _getArgs(ArgParser argParser, List<String> arguments) {
+  final ArgResults args = argParser.parse(arguments);
+
+  gcloudProjectId = args[gcloudProjectIdFlag];
+  gcloudProjectVersion = args[gcloudProjectVersionFlag];
+
+  if (gcloudProjectId == null) {
+    stderr.write('--$gcloudProjectIdFlag must be defined\n');
+    return false;
+  }
+
+  if (gcloudProjectVersion == null) {
+    stderr.write('--$gcloudProjectVersionFlag must be defined\n');
+    return false;
+  }
+
+  return true;
+}
+
+/// Build app_flutter for web.
+Future<bool> _buildFlutterWebApp() async {
+  final Process process = await Process.start(
+      'flutter', <String>['build', 'web'],
+      workingDirectory: '../app_flutter');
+  await stdout.addStream(process.stdout);
+
+  return await process.exitCode == 0;
+}
+
+/// Copy the built project from app_flutter to this app_dart project.
+Future<bool> _copyFlutterApp() async {
+  final ProcessResult result =
+      await Process.run('cp', <String>['-r', '../app_flutter/build', 'build']);
+
+  return result.exitCode == 0;
+}
+
+/// Run the Google Cloud CLI tool to deploy to [gcloudProjectId] under 
+/// version [gcloudProjectVersion].
+Future<bool> _deployToAppEngine() async {
+  stdout.write('Deploying to AppEngine\n');
+  /// The Google Cloud deployment command is an interactive process. It will
+  /// print out what it is about to do, and ask for confirmation (Y/n).
+  final Process process = await Process.start(
+    'gcloud',
+    <String>[
+      'app',
+      'deploy',
+      '--project',
+      gcloudProjectId,
+      '--version',
+      gcloudProjectVersion,
+      '--no-promote',
+      '--no-stop-previous-version',
+    ],
+  );
+
+  await process.stderr.pipe(stdout);
+  await process.stdout.pipe(stdout);
+
+  // So this is probably not the best way to do it
+  await Future<void>.delayed(const Duration(seconds: 15), () => process.stdin.write('y\n'));
```

Comment:
How about `echo 'y' | gcloud app deploy ...`?

Reply:
That's a much better way of thinking about it. I updated it so it pipes the stdin of the user to this process. That way  the user can confirm/decline the deployment themselves.

Reply:
However, do you know if there's any way to get around this lint ignore? https://github.com/flutter/cocoon/pull/471/commits/ed09516aa654446eafb8d37204f56b478daa6cea#diff-48a39cddbe6cd5ba24449d014cd75722R79

When I await, it will stall (I assume because stdin never ever finishes piping information, which makes sense).

Reply:
Jonah recommended `unawaited` from pedantic, and that worked.